scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt

```diff
@@ -1,2503 +1,2569 @@
+      .isEqualTo(
+        """
+        $stringFilePath:1: $subclassedActivityErrorMessage
+        $wikiReferenceNote
+        """.trimIndent()
+      )
+  }
+
+  @Test
+  fun testFileContent_subclassedAppCompatActivity_fileContentIsNotCorrect() {
+    val requiredContent = "decorateWithScreenName(TEST_ACTIVITY)"
+    val prohibitedContent = "class SomeActivity: AppCompatActivity() {}"
+    tempFolder.newFolder("testfiles", "app", "src", "main")
+    val stringFilePath = "app/src/main/SomeActivity.kt"
+    tempFolder.newFile("testfiles/$stringFilePath").writeText(prohibitedContent + requiredContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(REGEX_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString().trim())
+      .isEqualTo(
+        """
+        $stringFilePath:1: $subclassedAppCompatActivityErrorMessage
+        $wikiReferenceNote
+        """.trimIndent()
+      )
+  }
+
+  @Test
+  fun testFileContent_subclassedDialogFragment_fileContentIsNotCorrect() {
+    val prohibitedContent = "class SomeDialogFragment: DialogFragment() {}"
+    tempFolder.newFolder("testfiles", "app", "src", "main")
+    val stringFilePath = "app/src/main/SomeDialogFragment.kt"
+    tempFolder.newFile("testfiles/$stringFilePath").writeText(prohibitedContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(REGEX_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString().trim())
+      .isEqualTo(
+        """
+        $stringFilePath:1: $subclassedDialogFragmentErrorMessage
+        $wikiReferenceNote
+        """.trimIndent()
+      )
+  }
+
+  @Test
+  fun testFileContent_activityDeclarationInManifest_withConfigChanges_fileContentIsNotCorrect() {
+    val prohibitedContent =
+      """
+        <?xml version="1.0" encoding="utf-8"?>
+        <manifest package="org.oppia.android">
+          <application android:name=".app.application.OppiaApplication">
+            <activity
+              android:name=".app.ExampleActivity"
+              android:configChanges="orientation" />
+          </application>
+        </manifest>
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main")
+    val stringFilePath = "app/src/main/AndroidManifest.xml"
+    tempFolder.newFile("testfiles/$stringFilePath").writeText(prohibitedContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(REGEX_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString().trim())
+      .isEqualTo(
+        """
+        $stringFilePath:6: $androidActivityConfigChangesErrorMessage
+        $wikiReferenceNote
+        """.trimIndent()
+      )
+  }
+
+  @Test
+  fun testFileContent_activityDeclarationInManifest_withFirebaseCrashlyticsEnabled_fileContentIsNotCorrect() {

```

Comment:
New tests here and below 
testFileContent_activityDeclarationInManifest_withFirebaseCrashlyticsEnabled_fileContentIsNotCorrect()
testFileContent_activityDeclarationInManifest_withFirebaseAnalyticsEnabled_fileContentIsNotCorrect()